### PR TITLE
[camb]add w8a8 support

### PR DESCRIPTION
**Pull request by JackWeiw**

### CODE REVIEW

File: dlinfer/vendor/camb/camb_ops.py [hunk 1]
```diff
@@ -368,35 +376,119 @@ def fused_moe(
     return out
 
 
+def _process_input_dim(x: Tensor, scale: Optional[Tensor] = None):
+    """
+    NOTE: Since torch_mlu_ops matmul kernels requires input to be tow dimension.
+    we need to reshape the input tensor to 2D tensor if it is 3D tensor.
+    """
+    bsz, seq_len = None, None
+    if x.dim() == 3:
```
Comment (jinminxi104) [inline dlinfer/vendor/camb/camb_ops.py:385]: please merge dim==3 and dim==2. just reshape to (bsz * seq_len, -1) for both dim3 and 2

File: dlinfer/vendor/camb/camb_ops.py [hunk 2]
```diff
@@ -368,35 +376,119 @@ def fused_moe(
...
+
+
+@register_ops(vendor_ops_registry)
+def dynamic_quant(
+    x: Tensor, quant_dtype: torch.dtype, quant_granularity: str = "PER_TOKEN"
+):
+    assert quant_dtype == torch.int8
+    assert quant_granularity == "PER_TOKEN"
+    if x.shape[-1] in smooth_dic:
+        smooth = smooth_dic[x.shape[-1]]
```
Comment (jinminxi104) [inline dlinfer/vendor/camb/camb_ops.py:424]: could u explain more about smooth_dic

**Review by jinminxi104 — CHANGES_REQUESTED** [review]

File: dlinfer/vendor/camb/__init__.py [hunk 3]
```diff
@@ -1,3 +1,6 @@
 import torch
 
 from . import pytorch_patch, camb_ops
+
+# TODO. weitao: camb torch-mlu-ops-v1.2.0 per_token_smooth_quantize need smooth_vec
+SMOOTH_VEC = torch.ones(8000, dtype=torch.float32, device="mlu")
```
Comment (jinminxi104) [inline dlinfer/vendor/camb/__init__.py:6]: please change to 8*1024

File: dlinfer/vendor/camb/camb_ops.py [hunk 4]
```diff
@@ -6,6 +6,8 @@
 from dlinfer.utils.registry import register_ops
 from dlinfer.utils.type_annotation import Tensor, Optional, Sequence, Tuple
 
+from .__init__ import SMOOTH_VEC
```
Comment (jinminxi104) [inline dlinfer/vendor/camb/camb_ops.py:9]: `from .__init__`   ==>   `from . `

File: dlinfer/vendor/camb/camb_ops.py [hunk 5]
```diff
@@ -368,35 +374,120 @@ def fused_moe(
...
+def dynamic_quant(
+    x: Tensor, quant_dtype: torch.dtype, quant_granularity: str = "PER_TOKEN"
+):
+    assert quant_dtype == torch.int8
+    assert quant_granularity == "PER_TOKEN"
+    global SMOOTH_VEC
+    if x.shape[-1] <= SMOOTH_VEC.shape[0]:
+        smooth = SMOOTH_VEC[: x.shape[-1]]
+    else:
+        smooth = torch.ones(x.shape[-1], dtype=torch.float32, device=x.device)
```
Comment (jinminxi104) [inline dlinfer/vendor/camb/camb_ops.py:425]: Don't create a new one locally, update the global SMOOTH_VEC. and round up to 2^N K.

**Review by jinminxi104 — CHANGES_REQUESTED** [review]

File: dlinfer/vendor/camb/camb_ops.py (hunk 5 above)
Comment (JackWeiw) [reply dlinfer/vendor/camb/camb_ops.py]: Done

File: dlinfer/vendor/camb/__init__.py (hunk 3 above)
Comment (JackWeiw) [reply dlinfer/vendor/camb/__init__.py]: Done
